core/src/session.rs

```diff
@@ -14,22 +14,33 @@ use rkyv::{Archive, Deserialize, Serialize};
 
 use dusk_plonk::prelude::*;
 
+/// The struct defining a Citadel session, a set of public values shared
+/// on-chain that represent the use of a service by an unknown user
 #[cfg_attr(
     feature = "rkyv-impl",
     derive(Archive, Serialize, Deserialize),
     archive_attr(derive(bytecheck::CheckBytes))
 )]
 #[derive(Debug)]
 pub struct Session {
+    /// The hash of the public key of the SP plus a nonce
```

Comment:
If you are not adding the public key to the nonce, I would avoid the word "plus". I suggest to rephrase it: The hash of the public key and a nonce (like you did in [line 29](https://github.dev/dusk-network/citadel/blob/46af66bc4e0d79a0c3251a6c4a9de8456a943035/core/src/session.rs#L29-L30)).